### PR TITLE
TST: dropping of nuisance columns for groupby ops #38815

### LINKED ISSUE #38815: TST: dropping of nuisance columns for groupby ops
xref #38774 

As far as I can tell, for groupby we only have two tests for dropping nuisance columns and these only involve sum/mean. A more robust and parameterized test over all the groupby ops that drop nuisance columns should be added.

### PR DESCRIPTION
**Pull request by horaceklai**
Add more aggregation functions for nuisance tests

- [ x] closes #38815
- [x ] tests added / passed
- [x ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -873,11 +868,21 @@ def test_omit_nuisance(df):
         grouped.agg(lambda x: x.sum(0, numeric_only=False))
 
 
-def test_omit_nuisance_sem(df):
-    # GH 38774 - sem should work with nuisance columns
+@pytest.mark.parametrize(
```
Comment (jreback) [inline pandas/tests/groupby/test_groupby.py:871]: can you use this fixture: all_numeric_reductions

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/test_groupby.py (hunk 1 above)
Comment (horaceklai) [reply pandas/tests/groupby/test_groupby.py:871]: @jreback do you mean I should add a fixture with that name, all_numeric_reductions? I cannot find that fixture inside the groupby folder

Comment (jreback) [reply pandas/tests/groupby/test_groupby.py:871]: its in the top level

Comment (horaceklai) [reply pandas/tests/groupby/test_groupby.py:871]: I found it. It is defined as 
`_all_numeric_reductions = [
    "sum",
    "max",
    "min",
    "mean",
    "prod",
    "std",
    "var",
    "median",
    "kurt",
    "skew",
]`

However, max and min do not drop column B which has dtype object. Also 'DataFrameGroupBy' object has no attribute 'kurt'

**Comment by jreback** [comment]
also pls merge master

File: pandas/tests/groupby/test_groupby.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/groupby/test_groupby.py:874]: can you add *another* test and parameterize on min/max that they *don't* drop (and put a comment that links to this issue)

**Comment by jreback** [comment]
https://github.com/pandas-dev/pandas/pull/43674/checks?check_run_id=3771738681
need sto assert this warning happens (this is pretty new btw)

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -876,14 +871,45 @@ def test_omit_nuisance(df):
...
+    ["max", "min"],
+)
+def test_keep_nuisance_agg(df, agg_function):
+    # GH 38815
     grouped = df.groupby("A")
-    result = grouped.sem()
-    expected = df.loc[:, ["A", "C", "D"]].groupby("A").sem()
+    result = getattr(grouped, agg_function)().columns
+    expected = df.loc[:, ["B", "C", "D"]].columns
+    np.testing.assert_array_equal(result, expected)
```
Comment (jreback) [inline pandas/tests/groupby/test_groupby.py:883]: *never* use `np.testing`, in this case use `tm.assert_frame_equal` (don'tjust compare the columns)

 I think we have a rule which should fail tests about this cc @MarcoGorelli

File: pandas/tests/groupby/test_groupby.py [hunk 3]
```diff
@@ -876,14 +871,45 @@ def test_omit_nuisance(df):
...
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    "agg_function",
+    ["skew"],
+)
+def test_omit_nuisance_warnings(df, agg_function):
+    # GH 38815
+    with tm.assert_produces_warning(
```
Comment (jreback) [inline pandas/tests/groupby/test_groupby.py:904]: what is this warning? why are you adding extra options?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/test_groupby.py (hunk 2 above)
Comment (MarcoGorelli) [reply pandas/tests/groupby/test_groupby.py:883]: Yeah I thought there was one - currently on holiday, will check next week when I'm back

File: pandas/tests/groupby/test_groupby.py (hunk 3 above)
Comment (horaceklai) [reply pandas/tests/groupby/test_groupby.py:904]: This was the warning that you wanted to be asserted. I took away the parameterization now

https://github.com/pandas-dev/pandas/pull/43674/checks?check_run_id=3771738681
need sto assert this warning happens (this is pretty new btw)

File: pandas/tests/groupby/test_groupby.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/groupby/test_groupby.py:883]: @horaceklai if you can change this one as well

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @horaceklai

**Pull request merged**